sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -425,7 +425,7 @@ object SQLConf {
     .createWithDefault(true)
 
   val ADAPTIVE_EXECUTION_SKEWED_JOIN_ENABLED =
-    buildConf("spark.sql.adaptive.optimizeSkewedJoin.enabled")
+    buildConf("spark.sql.adaptive.optimizeSkewedJoin")
```

Comment:
`optimizeSkewedJoin` is used as a namespace and has sub-configs. how about `spark.sql.adaptive.skewedJoinOptimization.enabled`?

cc @JkSelf @maryannxue

Reply:
+1 for `spark.sql.adaptive.skewedJoinOptimization.enabled` to apply the naming guidelines.